crates/nu-parser/src/parser.rs

```diff
@@ -4687,6 +4703,7 @@ pub fn parse_expression(
     working_set: &mut StateWorkingSet,
     spans: &[Span],
     expand_aliases_denylist: &[usize],
+    is_subexpression: bool,
```

Comment:
It feels kind of unfortunate that we need to track this through larger parts of the parser for this (certainly outside of my core expertise)